nncf/nncf_network.py

```diff
@@ -504,10 +504,14 @@ def _set_nncf_wrapped_model(self, value):
     def get_clean_shallow_copy(self) -> 'NNCFNetwork':
         # WARNING: Will reset pre- and post-ops of the underlying model. Use save_nncf_module_additions
         # and load_nncf_module_additions to preserve these, or temporary_clean_view().
-        return NNCFNetwork(self.get_nncf_wrapped_model(), self.input_infos,
-                           self._user_dummy_forward_fn, self._wrap_inputs_fn,
-                           self.scopes_without_shape_matching, self.ignored_scopes, self.target_scopes,
-                           reset=True)
+        from nncf.utils import save_module_training_state, load_module_training_state
+        saved_state = save_module_training_state(self)
+        model_copy = NNCFNetwork(self.get_nncf_wrapped_model(), self.input_infos,
+                    self._user_dummy_forward_fn, self._wrap_inputs_fn,
+                    self.scopes_without_shape_matching, self.ignored_scopes, self.target_scopes,
+                    reset=True)
+        load_module_training_state(model_copy, saved_state)
```

Comment:
In our examples we have Inception model, which has different forward method depending on training state

Reply:
I am sorry, I was mistaken. In out example ICNET is a model that has different forward pass in training mode.

https://github.com/openvinotoolkit/nncf/blob/3722466085f7202da34fe6539b20f0aef25bc6da/examples/common/models/segmentation/icnet.py#L211-L213

https://github.com/openvinotoolkit/nncf/blob/3722466085f7202da34fe6539b20f0aef25bc6da/examples/common/models/segmentation/icnet.py#L312-L321

Reply:
> Also, why not make it a context manager as well?
@vanyalzr using context manager is not convenient here. Because we create a copy of a model while load and save an original model state. Therefore, we have two instances that need to give as arguments to this context manager, this doesn't make this context manager usable in any other part of the code.